Detector/DetFCChhTrackerTkLayout/src/TkLayoutConicalEndcap_Geo.cpp

```diff
@@ -14,148 +13,171 @@ using DD4hep::Geometry::PlacedVolume;
 
 namespace det {
 static DD4hep::Geometry::Ref_t createTkLayoutTrackerConicalEndcap(DD4hep::Geometry::LCDD& lcdd,
-                                                           DD4hep::XML::Handle_t xmlElement,
-                                                           DD4hep::Geometry::SensitiveDetector sensDet) {
+                                                                  DD4hep::XML::Handle_t xmlElement,
+                                                                  DD4hep::Geometry::SensitiveDetector sensDet) {
   // shorthands
   DD4hep::XML::DetElement xmlDet = static_cast<DD4hep::XML::DetElement>(xmlElement);
   Dimension dimensions(xmlDet.dimensions());
   double l_overlapMargin = 0.01;
 
   // get sensitive detector type from xml
   DD4hep::XML::Dimension sdTyp = xmlElement.child(_Unicode(sensitive));  // retrieve the type
-  sensDet.setType(sdTyp.typeStr());  // set for the whole detector
+  sensDet.setType(sdTyp.typeStr());                                      // set for the whole detector
 
   // definition of top volume
   std::string detName = xmlDet.nameStr();
   DetElement worldDetElement(detName, xmlDet.id());
+  DetElement posEcapDetElement(worldDetElement, "posEcap", 0);
+  DetElement negEcapDetElement(worldDetElement, "negEcap", 1);
 
   Acts::ActsExtension::Config actsBarrelConfig;
-  actsBarrelConfig.isEndcap             = true;
+  actsBarrelConfig.isEndcap = true;
   Acts::ActsExtension* worldDetExt = new Acts::ActsExtension(actsBarrelConfig);
   worldDetElement.addExtension<Acts::IActsExtension>(worldDetExt);
```

Comment:
The extension needs to be added to the two endcap DetElements posEcapDetElement and negEcapDetElement

Reply:
That's true, I will try to substitute this by a clone of the DetElement, this should copy the Extensions as well.

Reply:
What is exactly the advantage of deep copying the DetElements, from creating them directly? Since the clone method was not tested for some time in DD4hep and we are not sure if the hierachies of the DetElements and their extensions are copied correctly, I am not sure if this is the way to go.